src/app.controller.ts

```diff
@@ -1,13 +1,96 @@
-import { Controller, } from '@nestjs/common';
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Headers,
+  HttpException,
+  Param,
+  Post,
+} from '@nestjs/common';
 import { HttpService } from '@nestjs/axios';
 import { ConfigService } from '@nestjs/config';
+import { ApiHeader, ApiOperation, ApiParam } from '@nestjs/swagger';
+import { catchError, concat, map, mergeMap, toArray } from 'rxjs';
+import { SqlReport } from './sql-report';
+import { QueryBody } from './query-body.dto';
 
 @Controller()
 export class AppController {
-
+  private dbUrl = this.configService.get('DATABASE_URL');
+  private queryUrl = this.configService.get('QUERY_URL');
+  private schemaDocId = this.configService.get('SCHEMA_CONFIG_ID');
   constructor(
     private http: HttpService,
     private configService: ConfigService,
+  ) {}
+
+  // TODO also support cookie auth? Not really required with Keycloak
+  @ApiOperation({
+    description: `Get the results for the report with the given ID. User needs 'read' access for the requested report entity.`,
+  })
+  @ApiParam({ name: 'id', description: '(full) ID of the report entity' })
+  @ApiParam({ name: 'db', example: 'app', description: 'name of database' })
+  @ApiHeader({
+    name: 'Authorization',
+    required: false,
+    description: 'request needs to be authenticated',
+  })
+  @Post(':db/:id')
```

Comment:
shall we put some fixed prefix into the route, like "query/:db/:id"? 
to be prepared if there will be more endpoints; and also the OpenAPI docs look very non-descript with the route just containing parameters ;-)

Reply:
I was a little thinking about the reverse proxy that we will use in the frontend. So the url would already look like `/query/app/ReportConfig:1`. So another query in there would look a little weird. But if you have another idea to structure the URL, that would be great.